avocado/core/test.py

```diff
@@ -777,3 +777,17 @@ class ReplaySkipTest(SkipTest):
     """
 
     _skip_reason = "Test skipped due to a job replay filter!"
+
+
+class TestLoaderError(Test):
+    """
+    Handle when test loader faces an exception (i.e. python code error).
+    """
+
+    def __init__(self, *args, **kwargs):
+        exception = kwargs.pop('exception')
+        Test.__init__(self, *args, **kwargs)
+        self.exception = exception
+
+    def test(self):
+        raise exceptions.TestLoaderException(self.exception)
```

Comment:
I don't see much benefit in having extra exception class for this kind of failure. Pure `self.fail(...)` would be IMO sufficient.